.github/workflows/codeql.yml

```diff
@@ -0,0 +1,72 @@
+name: "CodeQL Advanced"
+
+on:
+  # We are checking master and book-v4 because book-v4 has always the code
+  # from the latest release
+  push:
+    branches: ["master", "book-v4"]
+  pull_request:
+    branches: ["master", "book-v4"]
+  schedule:
+    - cron: '30 20 * * 1'  # Runs a periodic scan every Monday at 8:30 PM
+
+jobs:
+  analyze:
+    name: Analyze (${{ matrix.language }})
+    runs-on: ubuntu-latest
+    permissions:
+      security-events: write
+      packages: read
+      actions: read
+      contents: read
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - language: go
+            build-mode: autobuild
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.22'
+
+      - name: Install Kubebuilder tools
+        run: |
+          go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.16.4
+          go install sigs.k8s.io/kustomize/kustomize/v5@v5.5.0
+
+      - name: Build and install Kubebuilder CLI
+        run: make install
+
+      # Construa apenas o projeto project-v4 no diretório testdata
+      - name: Build project-v4 sample project
+        run: |
+          cd testdata/project-v4
+          go mod tidy
+          make all
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v3
+        with:
+          languages: ${{ matrix.language }}
+          build-mode: ${{ matrix.build-mode }}
+
+      - if: matrix.language == 'go' && matrix.build-mode == 'manual'
```

Comment:
not sure when this condition is true (`manual` build mode)

Reply:
Thx good point, I think we can simplify it !